Gems/Multiplayer/Code/Source/AutoGen/NetworkHierarchyRootComponent.AutoComponent.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+
+<Component
+    Name="NetworkHierarchyRootComponent"
```

Comment:
I haven't had a chance to dig in too deeply yet, but what happens when roots change at runtime with these components? For example, a weapon that I'm holding suddenly becomes unrooted when I drop it, or my player suddenly is no longer the root if I jump on a moving platform or enter a vehicle.. stuff like this. Do all network entities potentially have both a child and root component on them or do we need to deactivate/activate to change parenting or is something else going on here?

Reply:
Also there's a bunch of potentially vestigial stuff lying around NetworkTransformComponent that you should probably delete.. I think there's a parent entity Id in there at least that's probably obsolete with these new components. Also how does this, or does it at all, integrate with Transform parenting?

Reply:
I'm aiming to pick up all parent-child changes from TransformComponent and replicate them over the network. So far it's looking good in unit tests and in minimal real testing in MultiplayerSample.